Improve email sharing

ekala opened the pull request:
- Users have to enter the security code in a captcha image before
  sending out the sharing email; added security measure. The captcha is served via the `captcha` module - https://github.com/ekala/kohana-captcha
- The email body is no longer free text and has been replaced with a
  standard message
- For limiting the no. of emails that a user can send, we'll probably have to extend this functionality to include tracking the no. of times an item is shared across the various networks?

69mb committed:
River public token + changes for river rss feed.

* Changes to allow the feed writer plugin to present the feed icon and
alternate links in the header.
* River now has a public access token.

ekala committed:
Improve email sharing

* Users have to enter the security code in a captcha image before
sending out the sharing email; added security measure

* The email body is no longer free text and has been replaced with a
standard message

ekala committed:
Improve email sharing

* Users have to enter the security code in a captcha image before
sending out the sharing email; added security measure

* The email body is no longer free text and has been replaced with a
standard message

ekala committed:
Add drop summary at the footer of the sharing dialog

* The summary is to serve as a visual cue as to what is being shared -
channel, author and title of the drop

ekala committed:
Add drop summary at the footer of the sharing dialog

* The summary is to serve as a visual cue as to what is being shared -
channel, author and title of the drop

69mb commented on modules/captcha:
```diff
@@ -0,0 +1 @@
+Subproject commit f45380bd7bb97bf4a5824056292b2258fe9efc10
```
Update this to the current master 17b1f5650bd329a315e88d6f07d449801db67d1d

69mb commented on application/classes/controller/drop/base.php:
```diff
@@ -246,24 +246,27 @@ public function action_share()
-				$mail_body = __(":body \n\nShared by :sender",
-				    array(':body' => $post['body'], ':sender' => $this->user->username));
+				$mail_body = __(":user has shared a drop with you via SwiftRiver\n\n:url",
+				    array(':sender' => $this->user->username, ':url' => $post['drop_url']));
```
:user is not set

69mb commented:
- The dialog needs visual feedback. Show the loading icon when the request is taking too long to also prevent multiple submissions.
- Improve the error message when there is an issue with the captcha or email.
- The correct headers are missing from the response returned with the captcha image so the same image is displayed every time.

ekala committed:
Email sharing improvements

* Visual feedback when a user clicks the "Send" button

* Improved error message when the recipient email address or security
code in the captcha is invalid

* Updated the captcha module to the current master; The captcha module
is now sending the correct headers in the response

ekala committed:
Email sharing improvements

* Visual feedback when a user clicks the "Send" button

* Improved error message when the recipient email address or security
code in the captcha is invalid

* Updated the captcha module to the current master; The captcha module
is now sending the correct headers in the response

ekala committed:
Merge pull request #231 from ushahidi/readme-dev

Add installation notes - Closes #230

ekala committed:
README update: Bash one-liner for creating the config files

ekala committed:
Merge branch 'email-dev' of github.com:ushahidi/SwiftRiver into email-dev

ekala committed:
Update rss submodule (plugins/rss) to current master

69mb commented:
+1
